desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,818 @@
+            L = ``grid.num_rho``, m = ``grid.num_theta``, and n = ``grid.num_zeta``.
+        data : dict[str, jnp.ndarray]
+            Data evaluated on ``grid``.
+            Must include names in ``Bounce2D.required_names``.
+        theta : jnp.ndarray
+            Shape (L, M, N).
+            DESC coordinates θ sourced from the Clebsch coordinates
+            ``FourierChebyshevSeries.nodes(M,N,L,domain=(0,2*jnp.pi))``.
+        N_B : int
+            Desired Chebyshev spectral resolution for |B|.
+        alpha : float
+            Starting field line poloidal label.
+        num_transit : int
+            Number of toroidal transits to follow field line.
+        quad : tuple[jnp.ndarray]
+            Quadrature points xₖ and weights wₖ for the approximate evaluation of an
+            integral ∫₋₁¹ g(x) dx = ∑ₖ wₖ g(xₖ). Default is 32 points.
+            For weak singular integrals, use ``chebgauss2`` from
+            ``desc.integrals.quad_utils``.
+            For strong singular integrals, use ``leggauss``.
+        automorphism : tuple[Callable] or None
+            The first callable should be an automorphism of the real interval [-1, 1].
+            The second callable should be the derivative of the first. This map defines
+            a change of variable for the bounce integral. The choice made for the
+            automorphism will affect the performance of the quadrature method.
+            For weak singular integrals, use ``None``.
+            For strong singular integrals, use
+            ``(automorphism_sin,grad_automorphism_sin)`` from
+            ``desc.integrals.quad_utils``.
+        Bref : float
+            Optional. Reference magnetic field strength for normalization.
+        Lref : float
+            Optional. Reference length scale for normalization.
+        is_reshaped : bool
+            Whether the arrays in ``data`` are already reshaped to the expected form of
+            shape (..., m, n) or (L, m, n). This option can be used to iteratively
+            compute bounce integrals one flux surface at a time, reducing memory usage
+            To do so, set to true and provide only those axes of the reshaped data.
+            Default is false.
+        check : bool
+            Flag for debugging. Must be false for JAX transformations.
+
+        """
+        errorif(grid.sym, NotImplementedError, msg="Need grid that works with FFTs.")
+        # Strictly increasing zeta knots enforces dζ > 0.
+        # To retain dℓ = (|B|/B^ζ) dζ > 0 after fixing dζ > 0, we require
+        # B^ζ = B⋅∇ζ > 0. This is equivalent to changing the sign of ∇ζ or [∂ℓ/∂ζ]|ρ,a.
+        # Recall dζ = ∇ζ⋅dR, implying 1 = ∇ζ⋅(e_ζ|ρ,a). Hence, a sign change in ∇ζ
+        # requires the same sign change in e_ζ|ρ,a to retain the metric identity.
+        warnif(
+            check and kwargs.pop("warn", True) and jnp.any(data["B^zeta"] <= 0),
+            msg="(∂ℓ/∂ζ)|ρ,a > 0 is required. Enforcing positive B^ζ.",
```

Comment:
how is positive B^zeta enforced?

Reply:
I just take abs(B^zeta) in line 428; same as `Bounce1D`